### ISSUE TITLE
warning  libmamba Added empty dependency for problem type SOLVER_RULE_UPDATE Could not solve for environment specs The following package could not be installed └─ segment-geospatial is installable and it requires    └─ pycocotools with the potential options

### ISSUE THREAD

**Issue by xingguangYan** [opengeos/segment-geospatial#251]
### Environment Information

-   samgeo version: latest version
-   Python version: 3.12
-   Operating System: Windows 10 Pro

### Description

Unable to install and run properly

### What I Did

I followed the steps below to install, but something went wrong in the fourth step.
conda create -n geo python
conda activate geo
conda install -c conda-forge mamba
mamba install -c conda-forge segment-geospatial

![image](https://github.com/opengeos/segment-geospatial/assets/87305660/156c7375-51a7-4360-affc-ff0686d88402)

```

(geo) C:\Users\xingguang>mamba install -c conda-forge segment-geospatial

Looking for: ['segment-geospatial']

warning  libmamba Cache file "C:\\Users\\xingguang\\miniconda3\\envs\\geo\\pkgs\\cache\\5afe41e9.json" was modified by another program
warning  libmamba Cache file "C:\\Users\\xingguang\\miniconda3\\envs\\geo\\pkgs\\cache\\09cdf8bf.json" was modified by another program
warning  libmamba Cache file "C:\\Users\\xingguang\\miniconda3\\envs\\geo\\pkgs\\cache\\59ba4880.json" was modified by another program
pkgs/main/win-64 (check zst)                        Checked  1.1s
warning  libmamba Cache file "C:\\Users\\xingguang\\miniconda3\\envs\\geo\\pkgs\\cache\\3e39a7aa.json" was modified by another program
pkgs/main/noarch (check zst)                        Checked  0.3s
warning  libmamba Cache file "C:\\Users\\xingguang\\miniconda3\\envs\\geo\\pkgs\\cache\\920c960f.json" was modified by another program
pkgs/r/win-64 (check zst)                           Checked  0.3s
warning  libmamba Cache file "C:\\Users\\xingguang\\miniconda3\\envs\\geo\\pkgs\\cache\\4ea078d6.json" was modified by another program
pkgs/r/noarch (check zst)                           Checked  0.3s
warning  libmamba Cache file "C:\\Users\\xingguang\\miniconda3\\envs\\geo\\pkgs\\cache\\5ca77eed.json" was modified by another program
pkgs/msys2/win-64 (check zst)                       Checked  0.3s
warning  libmamba Cache file "C:\\Users\\xingguang\\miniconda3\\envs\\geo\\pkgs\\cache\\c4a505b4.json" was modified by another program
pkgs/msys2/noarch (check zst)                       Checked  0.3s
pkgs/r/win-64                                      633.1kB @ 161.3kB/s  3.9s
pkgs/msys2/win-64                                   35.1kB @   7.2kB/s  0.9s
pkgs/msys2/noarch                                  103.0 B @  20.0 B/s  0.3s
pkgs/main/noarch                                   703.5kB @ 132.8kB/s  5.3s
conda-forge/noarch                                  13.7MB @ 797.3kB/s 17.2s
pkgs/r/noarch                                        2.1MB @  94.0kB/s 16.7s
pkgs/main/win-64                                     5.3MB @  47.6kB/s 1m:50.3s
conda-forge/win-64                                  21.0MB @ 152.8kB/s 2m:17.7s

Pinned packages:
  - python 3.12.*


warning  libmamba Added empty dependency for problem type SOLVER_RULE_UPDATE
Could not solve for environment specs
The following package could not be installed
└─ segment-geospatial is installable and it requires
   └─ pycocotools with the potential options
      ├─ pycocotools [2.0.4|2.0.6] would require
      │  └─ python >=3.10,<3.11.0a0 , which can be installed;
      ├─ pycocotools 2.0.4 would require
      │  └─ python >=3.7,<3.8.0a0 , which can be installed;
      ├─ pycocotools [2.0.4|2.0.6] would require
      │  └─ python >=3.8,<3.9.0a0 , which can be installed;
      ├─ pycocotools [2.0.4|2.0.6] would require
      │  └─ python >=3.9,<3.10.0a0 , which can be installed;
      └─ pycocotools 2.0.6 would require
         └─ python >=3.11,<3.12.0a0 , which can be installed.

```

```
![image](https://github.com/opengeos/segment-geospatial/assets/87305660/61e30287-c719-4895-8d61-e0a7f785cdd3)
![image](https://github.com/opengeos/segment-geospatial/assets/87305660/e72d5524-77fe-455f-8858-1ff5c88b5735)

Using cached annotated_types-0.6.0-py3-none-any.whl (12 kB)
Using cached mpmath-1.3.0-py3-none-any.whl (536 kB)
Building wheels for collected packages: color-operations
  Building wheel for color-operations (pyproject.toml): started
  Building wheel for color-operations (pyproject.toml): finished with status 'error'
Failed to build color-operations
Note: you may need to restart the kernel to use updated packages.
  error: subprocess-exited-with-error
  
  Building wheel for color-operations (pyproject.toml) did not run successfully.
  exit code: 1
  
  [46 lines of output]
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-cpython-312
  creating build\lib.win-amd64-cpython-312\color_operations
  copying color_operations\operations.py -> build\lib.win-amd64-cpython-312\color_operations
  copying color_operations\utils.py -> build\lib.win-amd64-cpython-312\color_operations
  copying color_operations\__init__.py -> build\lib.win-amd64-cpython-312\color_operations
  running egg_info
  writing color_operations.egg-info\PKG-INFO
  writing dependency_links to color_operations.egg-info\dependency_links.txt
  writing requirements to color_operations.egg-info\requires.txt
  writing top-level names to color_operations.egg-info\top_level.txt
  reading manifest file 'color_operations.egg-info\SOURCES.txt'
  adding license file 'LICENSE'
  writing manifest file 'color_operations.egg-info\SOURCES.txt'
  copying color_operations\colorspace.c -> build\lib.win-amd64-cpython-312\color_operations
  copying color_operations\colorspace.pyx -> build\lib.win-amd64-cpython-312\color_operations
  running build_ext
  building 'color_operations.colorspace' extension
  creating build\temp.win-amd64-cpython-312
  creating build\temp.win-amd64-cpython-312\Release
  creating build\temp.win-amd64-cpython-312\Release\color_operations
  "[C:\Program](file:///C:/Program) Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -I[C:\Users\xingguang\AppData\Local\Temp\pip-build-env-y5zpohyc\overlay\Lib\site-packages\numpy\core\include](file:///C:/Users/xingguang/AppData/Local/Temp/pip-build-env-y5zpohyc/overlay/Lib/site-packages/numpy/core/include) -I[C:\Users\xingguang\miniconda3\envs\geo\include](file:///C:/Users/xingguang/miniconda3/envs/geo/include) -I[C:\Users\xingguang\miniconda3\envs\geo\Include](file:///C:/Users/xingguang/miniconda3/envs/geo/Include) "-I[C:\Program](file:///C:/Program) Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-I[C:\Program](file:///C:/Program) Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" "-I[C:\Program](file:///C:/Program) Files (x86)\Windows Kits\8.1\include\shared" "-I[C:\Program](file:///C:/Program) Files (x86)\Windows Kits\8.1\include\um" "-I[C:\Program](file:///C:/Program) Files (x86)\Windows Kits\8.1\include\winrt" /Tccolor_operations/colorspace.c /Fobuild\temp.win-amd64-cpython-312\Release\color_operations/colorspace.obj -O2
  colorspace.c
  [c:\users\xingguang\appdata\local\temp\pip-build-env-y5zpohyc\overlay\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h](file:///C:/users/xingguang/appdata/local/temp/pip-build-env-y5zpohyc/overlay/lib/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h)(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
  color_operations/colorspace.c(7488): error C2039: 'curexc_traceback': is not a member of '_ts'
  [c:\users\xingguang\miniconda3\envs\geo\include\cpython/pystate.h](file:///C:/users/xingguang/miniconda3/envs/geo/include/cpython/pystate.h)(115): note: see declaration of '_ts'
  color_operations/colorspace.c(7491): error C2039: 'curexc_traceback': is not a member of '_ts'
  [c:\users\xingguang\miniconda3\envs\geo\include\cpython/pystate.h](file:///C:/users/xingguang/miniconda3/envs/geo/include/cpython/pystate.h)(115): note: see declaration of '_ts'
  color_operations/colorspace.c(9287): error C2039: 'ob_digit': is not a member of '_longobject'
  [c:\users\xingguang\miniconda3\envs\geo\include\cpython/longintrepr.h](file:///C:/users/xingguang/miniconda3/envs/geo/include/cpython/longintrepr.h)(87): note: see declaration of '_longobject'
  color_operations/colorspace.c(9342): error C2039: 'ob_digit': is not a member of '_longobject'
  [c:\users\xingguang\miniconda3\envs\geo\include\cpython/longintrepr.h](file:///C:/users/xingguang/miniconda3/envs/geo/include/cpython/longintrepr.h)(87): note: see declaration of '_longobject'
  color_operations/colorspace.c(9521): error C2039: 'ob_digit': is not a member of '_longobject'
  [c:\users\xingguang\miniconda3\envs\geo\include\cpython/longintrepr.h](file:///C:/users/xingguang/miniconda3/envs/geo/include/cpython/longintrepr.h)(87): note: see declaration of '_longobject'
  color_operations/colorspace.c(9576): error C2039: 'ob_digit': is not a member of '_longobject'
  [c:\users\xingguang\miniconda3\envs\geo\include\cpython/longintrepr.h](file:///C:/users/xingguang/miniconda3/envs/geo/include/cpython/longintrepr.h)(87): note: see declaration of '_longobject'
  color_operations/colorspace.c(9755): error C2039: 'ob_digit': is not a member of '_longobject'
  [c:\users\xingguang\miniconda3\envs\geo\include\cpython/longintrepr.h](file:///C:/users/xingguang/miniconda3/envs/geo/include/cpython/longintrepr.h)(87): note: see declaration of '_longobject'
  color_operations/colorspace.c(9810): error C2039: 'ob_digit': is not a member of '_longobject'
  [c:\users\xingguang\miniconda3\envs\geo\include\cpython/longintrepr.h](file:///C:/users/xingguang/miniconda3/envs/geo/include/cpython/longintrepr.h)(87): note: see declaration of '_longobject'
  color_operations/colorspace.c(10261): error C2039: 'ob_digit': is not a member of '_longobject'
  [c:\users\xingguang\miniconda3\envs\geo\include\cpython/longintrepr.h](file:///C:/users/xingguang/miniconda3/envs/geo/include/cpython/longintrepr.h)(87): note: see declaration of '_longobject'
  error: command '[C:\\Program](file:///C://Program) Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\cl.exe' failed with exit code 2
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for color-operations
ERROR: Could not build wheels for color-operations, which is required to install pyproject.toml-based projects
```

**Comment by hafzaal2** [comment]
I had the same issue. It was resolved by downgrading to Python 3.10.

**Comment by xingguangYan** [comment]
> I had the same issue. It was resolved by downgrading to Python 3.10.

Thank you, I will try.

**Comment by xingguangYan** [comment]
> I had the same issue. It was resolved by downgrading to Python 3.10.
Have you had this problem before?
File [~\miniconda3\envs\geo\Lib\site-packages\torchvision\extension.py:48](http://localhost:8888/~/miniconda3/envs/geo/Lib/site-packages/torchvision/extension.py#line=47), in _assert_has_ops()
     46 def _assert_has_ops():
     47     if not _has_ops():
---> 48         raise RuntimeError(
     49             "Couldn't load custom C++ ops. This can happen if your PyTorch and "
     50             "torchvision versions are incompatible, or if you had errors while compiling "
     51             "torchvision from source. For further information on the compatible versions, check "
     52             "https://github.com/pytorch/vision#installation for the compatibility matrix. "
     53             "Please check your PyTorch version with torch.__version__ and your torchvision "
     54             "version with torchvision.__version__ and verify if they are compatible, and if not "
     55             "please reinstall torchvision so that it matches your PyTorch install."
     56         )

RuntimeError: Couldn't load custom C++ ops. This can happen if your PyTorch and torchvision versions are incompatible, or if you had errors while compiling torchvision from source. For further information on the compatible versions, check https://github.com/pytorch/vision#installation for the compatibility matrix. Please check your PyTorch version with torch.__version__ and your torchvision version with torchvision.__version__ and verify if they are compatible, and if not please reinstall torchvision so that it matches your PyTorch install.
![image](https://github.com/opengeos/segment-geospatial/assets/87305660/86998023-2032-4231-a961-a16661e79201)

**Comment by hafzaal2** [comment]
Yes, I had the same issue. You can fix this by installing the correct version of Pytorch and Torchvision, as suggested. 
For me, "conda install pytorch==2.2.0 torchvision==0.17.0 torchaudio==2.2.0 pytorch-cuda=11.8 -c pytorch -c nvidia" worked great with GPU support. Based on your hardware, you can visit this page: "https://pytorch.org/get-started/previous-versions/" to determine the appropriate combination. 

After this, you may encounter another error, which can be fixed by "conda install anaconda::xarray"  
Good luck.
